src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,261 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final User INITIAL_USER_IN_STORAGE = new User(1L, "validLogin",
+            "validPassword", 18);
+    private static final int INITIAL_SIZE_OF_STORAGE = 1;
+    private static final String VALID_LOGIN_EXAMPLE = "user20";
+    private static final String VALID_PASSWORD_EXAMPLE = "password";
+    private static final int MIN_VALID_AGE = 18;
+    private static final String INVALID_LOGIN_EXAMPLE1 = "user1";
+    private static final String INVALID_LOGIN_EXAMPLE2 = "user";
+    private static final String INVALID_LOGIN_EXAMPLE3 = "u";
+    private static final String INVALID_LOGIN_EXAMPLE4 = "";
+    private static final String INVALID_LOGIN_EXAMPLE5 = "      ";
+    private static final String INVALID_PASSWORD_EXAMPLE1 = "12345";
+    private static final String INVALID_PASSWORD_EXAMPLE2 = "pass";
+    private static final String INVALID_PASSWORD_EXAMPLE3 = "1";
+    private static final String INVALID_PASSWORD_EXAMPLE4 = "";
+    private static final String INVALID_PASSWORD_EXAMPLE5 = "      ";
+
+    private static final int INVALID_AGE_EXAMPLE = 15;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.add(INITIAL_USER_IN_STORAGE);
+    }
+
+    @Test
+    void register_newUser_ok() {
+        User expected = new User(1L, VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        User actual = null;
+        try {
+            actual = registrationService.register(expected);
+        } catch (RegistrationException e) {
+            fail(e.getMessage());
+        }
+        assertEquals(expected, actual);
+        int expectedSizeOfStorage = INITIAL_SIZE_OF_STORAGE + 1;
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(expectedSizeOfStorage, actualSizeOfStorage);
```

Comment:
When you test creation of some objects, it is better to check if the object is in the  storage using method .get(). 
```suggestion
    void register_newUser_ok() {
        User expected = new User(1L, VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
        registrationService.register(expected);
        int expectedSizeOfStorage = INITIAL_SIZE_OF_STORAGE + 1;
        int actualSizeOfStorage = Storage.people.size();
        assertEquals(expectedSizeOfStorage, actualSizeOfStorage);
        User actual = storageDaoImpl.get(expected.getUsernam();
        assertEquals(expected, actual);
        
```